frontends/web/src/routes/device/bitbox02/upgradebutton.tsx

```diff
@@ -15,106 +15,82 @@
  * limitations under the License.
  */
 
-import { Component } from 'react';
-import { VersionInfo } from '../../../api/bitbox02';
-import { translate, TranslateProps } from '../../../decorators/translate';
-import { apiPost } from '../../../utils/request';
+import { useTranslation } from 'react-i18next';
+import { useState } from 'react';
+import { VersionInfo, upgradeDeviceFirmware } from '../../../api/bitbox02';
 import { Dialog, DialogButtons } from '../../../components/dialog/dialog';
 import { Button } from '../../../components/forms';
 import { SettingsButton } from '../../../components/settingsButton/settingsButton';
 import { RedDot } from '../../../components/icon/icon';
 
-interface UpgradeButtonProps {
-    apiPrefix: string;
-    versionInfo?: VersionInfo;
+export type TProps = {
     asButton?: boolean;
+    deviceID: string;
+    versionInfo?: VersionInfo;
 }
 
-type Props = UpgradeButtonProps & TranslateProps;
+export const UpgradeButton = ({ asButton, deviceID, versionInfo }: TProps) => {
+  const { t } = useTranslation();
 
-interface State {
-    activeDialog: boolean;
-    confirming: boolean;
-}
+  const [activeDialog, setActiveDialog] = useState(false);
+  const [confirming, setConfirming] = useState(false);
 
-class UpgradeButton extends Component<Props, State> {
-  public readonly state: State = {
-    activeDialog: false,
-    confirming: false,
-  };
+  const abort = () => setActiveDialog(false);
 
-  private upgradeFirmware = () => {
-    this.setState({ confirming: true });
-    apiPost(this.props.apiPrefix + '/upgrade-firmware').then(() => {
-      this.setState({ confirming: false });
-      this.abort();
+  const upgradeFirmware = () => {
+    setConfirming(true);
+    upgradeDeviceFirmware(deviceID).then(() => {
+      setConfirming(false);
+      abort();
     });
   };
 
-  private abort = () => {
-    this.setState({ activeDialog: false });
-  };
+  if (!versionInfo || !versionInfo.canUpgrade) {
+    return null;
+  }
 
-  public render() {
-    const {
-      t,
-      versionInfo,
-      asButton,
-    } = this.props;
-    const {
-      activeDialog,
-      confirming,
-    } = this.state;
-    if (!versionInfo || !versionInfo.canUpgrade) {
-      return null;
-    }
-    return (
-      <div>
-        {
-          asButton ? (
-            <Button primary onClick={() => this.setState({ activeDialog: true })}>
+  return (
+    <div>
+      {
+        asButton ? (
```

Comment:
since you are here anyway, could you remove the newline?

```suggestion
       { asButton ? (
```

Reply:
ack